cmd/modelcmd/credentials.go

```diff
@@ -21,26 +21,60 @@ var (
 	ErrMultipleCredentials = errors.New("more than one credential detected")
 )
 
+// GetCredentialsParams contains parameters for the GetCredentials function.
+type GetCredentialsParams struct {
+	// Cloud is the cloud definition.
+	Cloud cloud.Cloud
+
+	// CloudName is the name of the cloud for which credentials are being
+	// obtained.
+	CloudName string
+
+	// CloudRegion is the name of the region that the user has specified.
+	// If this is empty, then GetCredentials will determine the default
+	// region, and return that.
+	CloudRegion string
+
+	// CredentialName is the name of the credential to get.
+	CredentialName string
+}
+
 // GetCredentials returns a curated set of credential values for a given cloud.
 // The credential key values are read from the credentials store and the provider
 // finalises the values to resolve things like json files.
 // If region is not specified, the default credential region is used.
 func GetCredentials(
 	ctx *cmd.Context,
 	store jujuclient.CredentialGetter,
-	region, credentialName, cloudName, cloudType string,
+	args GetCredentialsParams,
 ) (_ *cloud.Credential, chosenCredentialName, regionName string, _ error) {
 
 	credential, credentialName, defaultRegion, err := credentialByName(
-		store, cloudName, credentialName,
+		store, args.CloudName, args.CredentialName,
 	)
 	if err != nil {
 		return nil, "", "", errors.Trace(err)
 	}
 
-	regionName = region
+	regionName = args.CloudRegion
 	if regionName == "" {
 		regionName = defaultRegion
+		if regionName == "" && len(args.Cloud.Regions) > 0 {
+			// No region was specified, use the first region
+			// in the list.
+			regionName = args.Cloud.Regions[0].Name
+		}
+	}
+
+	cloudEndpoint := args.Cloud.Endpoint
+	cloudIdentityEndpoint := args.Cloud.IdentityEndpoint
+	if regionName != "" {
```

Comment:
From the reading of above code, the possibility that regionName == "" is slim. Either we do not need this check or we need to do something for the case if it is "". No?